mysys/my_gethwaddr.c

```diff
@@ -106,7 +110,11 @@ my_bool my_gethwaddr(uchar *to)
     uint i;
     for (i= 0; res && i < ifc.ifc_len / sizeof(ifr[0]); i++)
     {
-#ifdef __linux__
+#if defined(_AIX)
```

Comment:
Instead of doing a full code path for AIX here, please instead add to the beginning of the file

#if defined(__linux___)
#define HWADDR_DATA_ ifr_hwaddr.sa_data
#elif defined (_AIX)
#define HWADDR_DATA_ ifr_hwaddr
#endif

Then you can instead do in the code:
res= memcpy_and_test(to, (uchar *)&ifr[i].HWADDR_DATA

And thus avoid duplicated code

Reply:
sure.